gnocchi/rest/api.py

```diff
@@ -1515,15 +1515,24 @@ def post(self, create_metrics=False):
         unknown_metrics = []
         unknown_resources = []
         body_by_rid = {}
+
+        attribute_filter = {"or": []}
+        for original_resource_id, resource_id in body:
+            names = list(body[(original_resource_id, resource_id)].keys())
+            attribute_filter["or"].append({"and": [
+                    {"=": {"resource_id": resource_id}},
+                    {"in": {"name": names}}]})
+
+        all_metrics = {}
```

Comment:
should this be pre-populated with all resource_id? i imagine if a resource does not have any metrics, the list_metrics below will never a key for resource and cause it to throw an error below

Reply:
all_metrics = collections.defaultdict(list), should fix that.

Reply:
yeah, true. that works even before.